packages/thirdweb/src/react/web/ui/prebuilt/Chain/icon.tsx

```diff
@@ -0,0 +1,169 @@
+
+  /**
+   * Optional query options for `useQuery`
+   */
+  queryOptions?: Omit<UseQueryOptions<string>, "queryFn" | "queryKey">;
+}
+
+/**
+ * This component tries to resolve the icon of a given chain, then return an image.
+ * @returns an <img /> with the src of the chain icon
+ *
+ * @example
+ * ### Basic usage
+ * ```tsx
+ * import { ChainProvider, ChainIcon } from "thirdweb/react";
+ *
+ * <ChainProvider chain={chain}>
+ *   <ChainIcon />
+ * </ChainProvider>
+ * ```
+ *
+ * Result: An <img /> component with the src of the icon
+ * ```html
+ * <img src="chain-icon.png" />
+ * ```
+ *
+ * ### Override the icon with the `iconResolver` prop
+ * If you already have the icon url, you can skip the network requests and pass it directly to the ChainIcon
+ * ```tsx
+ * <ChainIcon iconResolver="/ethereum-icon.png" />
+ * ```
+ *
+ * You can also pass in your own custom (async) function that retrieves the icon url
+ * ```tsx
+ * const getIcon = async () => {
+ *   const icon = getIconFromCoinMarketCap(chainId, etc);
+ *   return icon;
+ * };
+ *
+ * <ChainIcon iconResolver={getIcon} />
+ * ```
+ *
+ * ### Show a loading sign while the icon is being loaded
+ * ```tsx
+ * <ChainIcon loadingComponent={<Spinner />} />
+ * ```
+ *
+ * ### Fallback to a dummy image if the chain icon fails to resolve
+ * ```tsx
+ * <ChainIcon fallbackComponent={<img src="blank-image.png" />} />
+ * ```
+ *
+ * ### Usage with queryOptions
+ * ChainIcon uses useQuery() from tanstack query internally.
+ * It allows you to pass a custom queryOptions of your choice for more control of the internal fetching logic
+ * ```tsx
+ * <ChainIcon queryOptions={{ enabled: someLogic, retry: 3, }} />
+ * ```
+ *
+ * @component
+ * @chain
+ * @beta
+ */
+export function ChainIcon({
+  iconResolver,
+  loadingComponent,
+  fallbackComponent,
+  queryOptions,
+  client,
+  ...restProps
+}: ChainIconProps) {
+  const { chain } = useChainContext();
+  const iconQuery = useQuery({
+    queryKey: ["_internal_chain_icon_", chain.id] as const,
+    queryFn: async () => {
+      if (typeof iconResolver === "string") {
+        return iconResolver;
+      }
+      if (typeof iconResolver === "function") {
+        return iconResolver();
+      }
+      // Check if the chain object already has "icon"
+      if (chain.icon?.url) {
+        return chain.icon.url;
+      }
+      const possibleUrl = await getChainMetadata(chain).then(
+        (data) => data.icon?.url,
+      );
+      if (!possibleUrl) {
+        throw new Error("Failed to resolve icon for chain");
+      }
+      if (client) {
+        return resolveScheme({ uri: possibleUrl, client });
+      }
+      // If a thirdweb client is not passed, try to manually resolve the uri
+      if (possibleUrl.startsWith("ipfs://")) {
+        const cid = findIPFSCidFromUri(possibleUrl);
+        if (!cid) {
+          throw new Error(`Failed to resolve IPFS CID from ${possibleUrl}`);
+        }
+        return `https://ipfs.io/ipfs/${cid}`;
```

Comment:
Would not hardcode this. Keep it consistent with the rest of the SDK. If you want ipfs, you need a client

Reply:
Updated